src/entities/byron-validator.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+
+import type { Logger } from 'bunyan'
+import _ from 'lodash'
+import cbor from 'cbor'
+import bs58 from 'bs58'
+import blake from 'blakejs'
+// eslint-disable-next-line camelcase
+import { sha3_256 } from 'js-sha3'
+//import { injectable, decorate, inject } from 'inversify'
+import { helpers } from 'inversify-vanillajs-helpers'
+
+
+import type { TxType, UtxoInput } from '../blockchain/common'
+import {
+  StorageProcessor, NetworkConfig, Validator,
+} from '../interfaces'
+import SERVICE_IDENTIFIER from "../constants/identifiers";
+
+class ByronValidator implements Validator {
+
+  logger: Logger
+
+  storageProcessor: StorageProcessor
+
+  expectedNetworkMagic: number
+
+  constructor(
+    logger: Logger,
+    storageProcessor: StorageProcessor,
+    networkConfig: NetworkConfig,
+  ) {
+    this.logger = logger
+    this.storageProcessor = storageProcessor
+    this.expectedNetworkMagic = networkConfig.networkMagic()
+  }
+
+  async validateTx(txObj: TxType) {
+    try {
+      await this.validateTxWitnesses(txObj)
+      this.validateDestinationNetwork(txObj)
+      // TODO: more validation
+      return null
+    } catch (e) {
+      return e
+    }
+  }
+
+  async validateTxWitnesses({ id, inputs, witnesses }: TxType) {
```

Comment:
This file was more or less copied wholesale from the `api/txs.js` endpoint with the exception of some minor changes which you can see better by looking at the diff in the TxInputType -> AccountInput | UtxoInput commit